src/util/cmdline.cpp

```diff
@@ -20,175 +20,305 @@ cmdlinet::~cmdlinet()
 
 void cmdlinet::clear()
 {
-  options.clear();
+  vm.clear();
   args.clear();
 }
 
-bool cmdlinet::isset(char option) const
-{
-  int i;
-
-  i = getoptnr(option);
-  if(i < 0)
-    return false;
-  return options[i].isset;
-}
-
 bool cmdlinet::isset(const char *option) const
 {
-  int i;
-
-  i = getoptnr(option);
-  if(i < 0)
-    return false;
-  return options[i].isset;
+  return vm.count(option);
 }
 
-const char *cmdlinet::getval(char option) const
-{
-  int i;
 
-  i = getoptnr(option);
-  if(i < 0)
-    return (const char *)nullptr;
-  if(options[i].values.empty())
-    return (const char *)nullptr;
-  return options[i].values.front().c_str();
-}
 
-const std::list<std::string> &cmdlinet::get_values(char option) const
+const std::list<std::string> &cmdlinet::get_values(const char *option) const
 {
-  int i;
-
-  i = getoptnr(option);
-  assert(i >= 0);
-  return options[i].values;
+  auto src = vm[option].as<std::vector<std::string>>();
+  auto dest = new std::list<std::string>(src.begin(), src.end());
+  return *dest;
 }
 
 const char *cmdlinet::getval(const char *option) const
 {
-  int i;
-
-  i = getoptnr(option);
-  if(i < 0)
+  if(!vm.count(option))
     return (const char *)nullptr;
-  if(options[i].values.empty())
-    return (const char *)nullptr;
-  return options[i].values.front().c_str();
-}
-
-const std::list<std::string> &cmdlinet::get_values(const char *option) const
-{
-  int i;
-
-  i = getoptnr(option);
-  assert(i >= 0);
-  return options[i].values;
-}
-
-int cmdlinet::getoptnr(char option) const
-{
-  for(unsigned i = 0; i < options.size(); i++)
-    if(options[i].optchar == option)
-      return i;
-
-  return -1;
-}
-
-int cmdlinet::getoptnr(const char *option) const
-{
-  for(unsigned i = 0; i < options.size(); i++)
-    if(options[i].optstring == option)
-      return i;
+  auto &value = vm[option].value();
+  if(auto v = boost::any_cast<int>(&value))
+  {
+    const auto res = std::to_string(*v).c_str();
+    char *newstr = new char[strlen(res) + 1];
+    strcpy(newstr, res);
+    return newstr;
+  }
+  else if(auto v = boost::any_cast<std::string>(&value))
+    return v->c_str();
+  else if(auto v = boost::any_cast<std::vector<std::string>>(&value))
+    return v->front().c_str();
 
-  return -1;
+  else
+    return (const char *)nullptr;
 }
 
-bool cmdlinet::parse(int argc, const char **argv, const struct opt_templ *opts)
+void cmdlinet::parse(int argc, const char **argv)
 {
-  unsigned int i;
-
   clear();
 
```

Comment:
The help message is in `esbmc_parseoptions.cpp`, please copy the description from there.

Reply:
I can't understand what do you mean. Would you please explain this more?

Reply:
Sure, the help text is defined here: https://github.com/esbmc/esbmc/blob/master/src/esbmc/esbmc_parseoptions.cpp#L1639

I had a quick look and noticed that:
1. The defines for each architecture are missing (lines 1712-1727). 
2. The method `esbmc_parseoptionst::help` should now call the description generated by the `boost_options` object.

Reply:
Done.